src/java.base/share/classes/java/util/Locale.java

```diff
@@ -794,16 +786,46 @@ public Locale(String language, String country) {
      * any syntactic checks on the input.
      * </ul>
      *
+     * @deprecated Locale constructors have been deprecated. See <a href="#ObtainingLocale">
+     * Obtaining a Locale</a> for other options.
+     *
      * @param language An ISO 639 alpha-2 or alpha-3 language code, or a language subtag
      * up to 8 characters in length.  See the {@code Locale} class description about
      * valid language values.
      * @throws    NullPointerException thrown if argument is null.
      * @since 1.4
      */
+    @Deprecated(since="19")
     public Locale(String language) {
         this(language, "", "");
     }
 
+    /**
+     * Returns a {@code Locale} obtained from the given
+     * fields. Fields can take up to 4 arguments in the following order;
+     * {@code Locale.of(}<a href="#def_language">language</a>,
+     * <a href="#def_region">country</a>,
+     * <a href="#def_variant">variant</a>,
+     * <a href="#def_script">script</a>{@code )}.
+     * Each field can be an empty String, but cannot be {@code null}.
+     * If {@code fields} holds more than 4 arguments, the rest are ignored. If
+     * no arguments are given, {@link #ROOT} is returned.
+     *
+     * @param fields language, country, variant, and/or script, cannot be {@code null}.
+     * @return the {@code Locale} instance requested
+     * @throws    NullPointerException if any field argument is {@code null}.
+     * @since 19
+     */
+    public static Locale of(String... fields) {
+        return switch (fields.length) {
+            case 0 -> ROOT;
+            case 1 -> getInstance(fields[0], "", "", "", null);
+            case 2 -> getInstance(fields[0], "", fields[1], "", null);
+            case 3 -> getInstance(fields[0], "", fields[1], fields[2], null);
+            default -> getInstance(fields[0], fields[3], fields[1], fields[2], null);
```

Comment:
This should probably throw `IllegalArgumentException` when more than 4 fields are passed:
```suggestion
            case 4 -> getInstance(fields[0], fields[3], fields[1], fields[2], null);
            default -> throw new IllegalArgumentException(/* TODO: message */);
```

Reply:
Thanks for the suggestion. Will incorporate the exception in the spec/impl.